### ISSUE TITLE
"zebra" option in DataTable broken for g10 theme

### ISSUE THREAD

**Issue by jsonn** [carbon-design-system/carbon-components-svelte#966]
The `g10.css` file  as provided shipped doesn't distinguish between odd and even rows, both use the white background color. It does work correctly when using the all.css with `theme="g10"`.

**Comment by brunnerh** [comment]
The CSS is generated from the SCSS of the flagship package `carbon-components`, so such issues should probably be raised [there](https://github.com/carbon-design-system/carbon/issues).

**Comment by metonym** [comment]
I'm also seeing broken DataTable zebra styles when using g80, g90, and g100.

**Comment by brunnerh** [comment]
You are right, something is off. Made a [REPL here](https://svelte.dev/repl/d60eea2cf45c46948e02f4dc4fe8732b?version=3.44.3).

**Comment by metonym** [comment]
It seems that this bug results from how this library invokes each theme in the `css/*.scss` files (except for all.scss).

When setting a global theme variable, we should define it first:

```css
// Use the "g10" (Gray 10) Carbon theme
@import "carbon-components/scss/globals/scss/vendor/@carbon/themes/scss";

$carbon--theme: $carbon--theme--g10;
@include carbon--theme();

/** ... everything else */
```

When assigning the theme first, the zebra styles are applied correctly per theme (except for g80.css). The g80 issue is probably an upstream bug.

**Comment by metonym** [comment]
Hot fixed in [v0.51.1](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.51.1).

Opened an upstream issue regarding the Gray 80 bug: https://github.com/carbon-design-system/carbon/issues/10330
